automates/text_reading/src/main/scala/org/clulab/aske/automates/OdinActions.scala

```diff
@@ -357,6 +363,60 @@ class OdinActions(val taxonomy: Taxonomy, expansionHandler: Option[ExpansionHand
     resolved ++ woIt
   }
 
+  def resolveModelCoref(mentions: Seq[Mention], state: State = new State()): Seq[Mention] = {
+    val (models, nonModels) = mentions.partition(m => m.label == "ModelDescr")
+    val (theModel, modelNames) = models.partition(m => m.arguments.contains("model") && m.arguments("model").head.foundBy == "the/this_model" || m.arguments.contains("model") && m.arguments("model").head.foundBy == "model_pronouns")
+    val resolved: Seq[Mention] = theModel.map(m => replaceTheModel(mentions, m))
+    val resolved_filtered = resolved.filter(m => m.arguments("model").head.foundBy != "model_pronouns")
+        (resolved_filtered ++ modelNames ++ nonModels).distinct
+  }
+
+  def replaceTheModel(mentions: Seq[Mention], origModel: Mention): Mention = {
+    val previousModelInterval = returnPreviousModelInt(mentions, origModel)
+    val previousModelSntnce = returnPreviousModelSntnce(mentions, origModel)
+
+    if (previousModelInterval != Interval(0,0)) {
```

Comment:
Sorry, I should have caught this sooner, but below are a few comments about this method and the next two. You can try to adjust this based on the comments (we can meet to discuss) OR I can most likely get this done today or tomorrow. 

1) I think there might be a way to simplify this. You can have your previous model method return the mention you think is the previous Model mention----if you return that, that will have the info on both interval and sentence, so you won't have to have a separate sentence method in addition to the return interval method. This way you also won't have to construct a new text bound mention in line 379. To make sure you can account for cases where no previous model being available, this method can return an option (Option[Mention]) and not just a mention. 
2) I think it is possible for a previous model to not necessarily be in the previous sentence---it can easily be within the same sentence. I think we can just get all the model mentions available in the current and previous sentence and choose, let's say, the latest one of them.